tests/testthat/test-accuracy-published.R

```diff
@@ -0,0 +1,245 @@
+                                       model = model_power_a,
+                                       data = data_obs_a
+  )
+
+  ## (A) Actual table I
+  pat_info <- rbind(
+    "No. of patients" = c(3, 4, 5, 4, "-", "-", 2, "-", "-", "-"),
+    "No. of DLTs" = c(0, 0, 0, 0, "-", "-", 2, "-", "-", "-")
+  )
+  colnames(pat_info) <- c(1, 2.5, 5, 10, 15, 20, 25, 30, 40, 50)
+
+  tab1_a_act <- rbind(
+    "Skeleton (CRM)" = c(
+      0.01, 0.015, 0.020, 0.025, 0.03,
+      0.04, 0.05, 0.10, 0.17, 0.30,
+      0.45, 0.70, 0.80, 0.90, 0.95
+    )[1:10],
+    t(dose_rec_loss_a$probs[, c(6:7)])[, 1:10]
+  )
+
+
+  ## (B) Actual table I
+
+  data_obs_b <- crmPack::Data(
+    x = c(
+      rep(1, 3), rep(2.5, 4), rep(5, 5),
+      rep(10, 4), rep(25, 2)
+    ),
+    y = c(
+      rep(0, 3), rep(0, 4), rep(0, 5),
+      rep(0, 4), rep(1, 2)
+    ),
+    cohort = c(
+      rep(1, 3), rep(2, 4), rep(3, 5),
+      rep(4, 4), rep(7, 2)
+    ),
+    doseGrid = c(
+      1, 2.5, 5, 10, 15, 20, 25, 30,
+      40, 50
+    ),
+    ID = 1:18
+  )
+
+  model_power_b <- crmPack::OneParExpNormalPrior(
+    skel_probs = c(
+      0.063, 0.125, 0.188, 0.250, 0.313,
+      0.375, 0.438, 0.500, 0.563, 0.625
+    ),
+    dose_grid = c(
+      1, 2.5, 5, 10, 15, 20, 25, 30,
+      40, 50
+    ),
+    sigma2 = 1.34^2
+  )
+
+  post_samples_b <- crmPack::mcmc(data_obs_b, model_power_b, mcmc_options)
+
+  dose_rec_loss_b <- crmPack::nextBest(ncrm_loss,
+                                       doselimit = crmPack::maxDose(
+                                         increments_no,
+                                         data_obs_b
+                                       ),
+                                       samples = post_samples_b,
+                                       model = model_power_b,
+                                       data = data_obs_b
+  )
+
+  tab1_b_act <- rbind(
+    "Skeleton (CRM)" = c(
+      0.01, 0.015, 0.020, 0.025, 0.03,
+      0.04, 0.05, 0.10, 0.17, 0.30,
+      0.45, 0.70, 0.80, 0.90, 0.95
+    )[1:10],
+    # t(doseRec_loss_A[[4]])[c(7:8),1:10]),3)
+    t(dose_rec_loss_b$probs[, c(6:7)])[, 1:10]
+  )
+
+  ## (A)+ (B) Actual table I
+  tab1_act <- list(
+    "Posterior summaries (original skeleton)" = tab1_a_act,
+    "Posterior summaries (equidistant skeleton)" = tab1_b_act
+  )
+
+
+  ## Expected table I (Neuenschwander et al.)
+  tab_1_exp <- list()
+  tab1_names <- list(
+    c("Skeleton (CRM)", "Mean", "Std. dev."),
+    c(1, 2.5, 5, 10, 15, 20, 25, 30, 40, 50)
+  )
+
+  tab1 <- structure(list(X = c("No. of patients", "No. of DLTs", "A)
+                             Posterior summaries (original skeleton)",
+                               "Skeleton (CRM)", "Mean", "Std. dev.", "B)
+                             Posterior summaries (equidistant skeleton)",
+                               "Skeleton (CRM)", "Mean", "Std. dev."),
+                         X1 = c(3, 0, NA, 0.01, 0.069, 0.055, NA, 0.063,
```

Comment:
can you use more meaningful column names here?

Reply:
done